component/discovery/transformer/types.go

```diff
@@ -0,0 +1,107 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/grafana/regexp"
+)
+
+// Action is the relabelling action to be performed.
+type Action string
+
+// Regexp encapsulates the Regexp type from Grafana's
+// fork of the Go stdlib regexp package.
+// TODO (@tpaschalis) This encapsulation already exists in Prometheus' relabel.Regexp
+// so not sure whether to also move it here for now.
+type Regexp struct {
+	re       *regexp.Regexp
+	original string
+}
```

Comment:
nit: move the declarative for Regexp after all the Action-related things so the file reads more naturally from top down.